### ISSUE TITLE
BundleResource by default include to a nested directory, which can be confusing

### ISSUE THREAD

**Issue by chamons** [dotnet/maui#8612]
_From @PureWeen on Sat, 25 Jun 2022 15:53:08 GMT_

Moved from https://github.com/dotnet/maui/issues/8301

### Description

For firebase services the the "GoogleService-Info.plist" file build action has to be set as "BundleResource". The build action is not available in the visual studio selecten and also does not work when setting in the project file:

```
    <ItemGroup>
      <BundleResource  Include="Platforms\iOS\GoogleService-Info.plist" />
    </ItemGroup>
```

The following errror occurs when "Firebase.Core.App.Configure()" is called:
`ObjCRuntime.ObjCException: 'Objective-C exception thrown.  Name: com.firebase.core Reason: FirebaseApp.configure() could not find a valid GoogleService-Info.plist in your project. Please download one from https://console.firebase.google.com/.
Native stack trace:`

### Steps to Reproduce

1. dotnet new maui
2. Install nuget Xamarin.Firebase.iOS.CloudMessaging
3. Add GoogleService-Info.plist file to Platforms/iOS
4. Try to set build action
5. Call Firebase.Core.App.Configure(); in AppDelegate 

```
    public override bool FinishedLaunching(UIApplication application, NSDictionary launchOptions)
    {
        Firebase.Core.App.Configure();
        return base.FinishedLaunching(application, launchOptions);
    }
```

### Version with bug

6.0.400 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

iOS

### Affected platform versions

All

### Did you find any workaround?

Yes the configuration can be set in code

```
Firebase.Core.App.Configure(new Options("1:7..4:ios:8..7", "7..4")
{
	ApiKey = "A.._W.._8.._g",
	ClientId = "...apps.googleusercontent.com",
	BundleId = "<<BUNDLE-ID>>",
	ProjectId = "<<ProjectId>>"
});
```

_Copied from original issue xamarin/xamarin-macios#15347_

**Comment by chamons** [comment]
_From @chamons on Tue, 05 Jul 2022 15:46:48 GMT_

Have you looked inside the generated bundle to see if `GoogleService-Info.plist` is being copied anywhere?

Please attach a build log, ideally a binary build log, so we can look into what's going on.

**Comment by chamons** [comment]
_From @msftbot[bot] on Tue, 05 Jul 2022 15:46:59 GMT_

Hi @PureWeen. We have added the "need-info" label to this issue, which indicates that we have an open question for you before we can take further action. This issue will be closed automatically in 7 days if we do not hear back from you by then - please feel free to re-open it if you come back to this issue after that time.

**Comment by chamons** [comment]
_From @PureWeen on Tue, 05 Jul 2022 16:02:20 GMT_

> Have you looked inside the generated bundle to see if `GoogleService-Info.plist` is being copied anywhere?
> 
> Please attach a build log, ideally a binary build log, so we can look into what's going on.

@epsmae

**Comment by chamons** [comment]
_From @epsmae on Thu, 07 Jul 2022 07:28:53 GMT_

@PureWeen sorry for my late reply:
1. dotnet new maui
2. Added Platforms\iOS\GoogleService-Info.plist file
3. Add the following to csproj

```
    <ItemGroup>
      <BundleResource  Include="Platforms\iOS\GoogleService-Info.plist" />
    </ItemGroup>
```

Following the build log hope it helps (Note publish failed due to invalid publish profile)
[Verbose.log](https://github.com/xamarin/xamarin-macios/files/9061345/Verbose.log)
![screenshot_vs](https://user-images.githubusercontent.com/22543513/177717168-bc703f6a-d75a-4c9f-b64d-b604ee08a895.JPG)


```
	Line 87069: 1>    SmartCopy: 2022-07-07T00:19:51.2642423-07:00 - Creating output 'bin\Debug\net6.0-ios\ios-arm64\device-builds\iphone13.2-15.5\AppBundleSample.app\Platforms\iOS\GoogleService-Info.plist'...
	Line 111334: 1>            Platforms\iOS\GoogleService-Info.plist
```

**Comment by chamons** [comment]
_From @chamons on Thu, 07 Jul 2022 15:45:13 GMT_

It appears that file is being added to your bundle in `Platforms/iOS/GoogleService-Info.plist` 

```
 CopyingFile - Path: /Users/epsag/Library/Caches/Xamarin/mtbs/builds/AppBundleSample/14c0993197954ee38641618911ebdf8b1574d67cb49cb38b0dbf79d6e997d01a/bin/Debug/net6.0-ios/ios-arm64/device-builds/iphone13.2-15.5/AppBundleSample.app/Platforms/iOS/GoogleService-Info.plist
```

I assume that it needs to be in the root of your bundle?

**Comment by chamons** [comment]
_From @chamons on Thu, 07 Jul 2022 15:50:48 GMT_

There is some documentation [here](https://github.com/xamarin/xamarin-macios/blob/main/dotnet/BundleContents.md) on how BundleResource works on iOS, but I think you really want:

```
    <ItemGroup>
      <BundleResource  Include="Platforms\iOS\GoogleService-Info.plist" Link="GoogleService-Info.plist" />
    </ItemGroup>
```

To add the file to the root of your bundle.

**Comment by chamons** [comment]
_From @epsmae on Fri, 08 Jul 2022 15:38:25 GMT_

@chamons thanks, this works indeed. However I do not believe that any developer stumbles across this document.
My assumption is/was that the platform folder is already handled as root and i was not aware of the link property.

So I believe the bug or missing features is then that the build action is not shown in a maui project but you can select it in an net6-ios project.

**Comment by chamons** [comment]
_From @chamons on Fri, 08 Jul 2022 16:41:32 GMT_

I agree the behavior is less than ideal from a MAUI context, I'm going to transfer this bug over there for consideration.

**Comment by ohumidk** [comment]
Are there any updates for this?

What would a fix like this need? I have never done open source work but I would like to give it a shot if I can. I have been working on a solo project using Xamarin Forms for a few months now and I love how it's turning out. 

I love it so much and enjoy how great XF is as a tool that I am so onboard for MAUI but this is such a blocker. Firebase is such a great product for mobile devs and MAUI should take advantage of this powerful tool. 
I really see flutter as MAUI's competition and they really are jumping on Firebase so MAUI should too.

All that to say, MAUI is great. Keep up the amazing work. 👏

**Comment by valdetero** [comment]
Since VS didn't have an option for `BundleResource`, I used this workaround:
```
<ItemGroup Condition="$(TargetFramework.Contains('-ios'))">
	<MauiAsset Include="Platforms\iOS\GoogleService-Info.plist" LogicalName="GoogleService-Info.plist" />
</ItemGroup>
```

**Comment by navkar** [comment]
> _From @chamons on Thu, 07 Jul 2022 15:50:48 GMT_
> 
> There is some documentation [here](https://github.com/xamarin/xamarin-macios/blob/main/dotnet/BundleContents.md) on how BundleResource works on iOS, but I think you really want:
> 
> ```
>     <ItemGroup>
>       <BundleResource  Include="Platforms\iOS\GoogleService-Info.plist" Link="GoogleService-Info.plist" />
>     </ItemGroup>
> ```
> 
> To add the file to the root of your bundle.

This has actually worked for me :)

**Comment by Kirk-iOS** [comment]
> > _From @chamons on Thu, 07 Jul 2022 15:50:48 GMT_
> > There is some documentation [here](https://github.com/xamarin/xamarin-macios/blob/main/dotnet/BundleContents.md) on how BundleResource works on iOS, but I think you really want:
> > ```
> >     <ItemGroup>
> >       <BundleResource  Include="Platforms\iOS\GoogleService-Info.plist" Link="GoogleService-Info.plist" />
> >     </ItemGroup>
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > To add the file to the root of your bundle.
> 
> This has actually worked for me :)

This also worked for me.

**Comment by rolfbjarne** [comment]
Looks like this is a duplicate of https://github.com/dotnet/maui/issues/16734.

**Comment by Zhanglirong-Winnie** [comment]
Verified this issue with Visual Studio Enterprise 17.8.0 Preview 7.0. Can repro on iOS platform.
